### ISSUE TITLE
Ensure sanity checks on the `swapDescription`.flags

### ISSUE THREAD

**Issue by shealtielanz** [shieldify-security/Adapters-SPP#2]
## summary
The flags of the swapDescription determine a lot of things that will happen during a swap, causing incorrect swap types to be executed on 1inch.
## Details
The sponsors intention [here](https://github.com/shieldify-security/SPP-Adapters/issues/88#issuecomment-2079737207)
`However, my intention is that there are no partial fills because that would cause problems in UX. Imagine someone wants to receive upfront yield in USDC and only 10% of the order gets filled. Then the user gets 90% PSM and 10% USDC and would be confused why he didn´t get the USDC that he expected. He then needs to take an extra step to go to a DEX to sell the 90% PSM for USDC manually.`
`For that reason it must be an all or nothing swap`.

## Mitigation
Check the flags of the swapDescription

**Comment by PossumLabsCrypto** [comment]
Good point. Aside from UX inconvenience there would be no negative effect for the user without the check in case partial fills were allowed, so this is a valid low.
(We send remaining tokens back the user)

Check added
https://github.com/PossumLabsCrypto/Adapters/commit/f359adba3ecbc43b989ac85bfd085e1c691f2ef2

**Comment by shealtielanz** [comment]
Heyy @PossumLabsCrypto Thanks for the feedback.
I don't think the checks you made are complete.
Can you check and add.
```diff
+     uint256 private constant _REQUIRES_EXTRA_ETH = 1 << 1;
+     desc.flags & _REQUIRES_EXTRA_ETH != 0 revert ErrorsLib.InvalidSwap();
```

**Comment by PossumLabsCrypto** [comment]
Thank you for the review and suggestion.

I think the mentioned flag is related to native ETH trades when srcToken = ETH. The source token is always PSM in these swaps, so this check / flag is irrelevant. If there is a faulty flag for some reason, the transaction will just revert.

**Comment by shealtielanz** [comment]
The check if `src == Eth` is done after the flag is checked, though it may be irrelevant I think it is a safe to add.
```solidity

if (desc.flags & _REQUIRES_EXTRA_ETH != 0) {
            if (msg.value <= (srcETH ? desc.amount : 0)) revert RouterErrors.InvalidMsgValue();
        } else {
            if (msg.value != (srcETH ? desc.amount : 0)) revert RouterErrors.InvalidMsgValue();
        }

```
However, I agree with you, thanks for listening to my previous opinions.
